From d5497912a7e369d9de7703202c18634b7c5f5ca1 Mon Sep 17 00:00:00 2001
From: "andy.huang" <andy.huang@thales-cetca.com>
Date: Mon, 17 Sep 2018 17:38:58 +0800
Subject: [PATCH 16/16] add the hw wdog config

---
 drivers/watchdog/Kconfig  | 6 ++++++
 drivers/watchdog/Makefile | 3 +++
 2 files changed, 9 insertions(+)

diff --git a/drivers/watchdog/Kconfig b/drivers/watchdog/Kconfig
index 37460cd..6aa0696 100644
--- a/drivers/watchdog/Kconfig
+++ b/drivers/watchdog/Kconfig
@@ -37,6 +37,12 @@ config WATCHDOG_CORE
 	  and gives them the /dev/watchdog interface (and later also the
 	  sysfs interface).
 
+config HW_WATCHDOG
+        bool "WatchDog for TCA HW"
+        help
+          Say Y here if you want to use the new watchdog  driver.
+          This driver provides for TCA CIDU platform.
+
 config WATCHDOG_NOWAYOUT
 	bool "Disable watchdog shutdown on close"
 	help
diff --git a/drivers/watchdog/Makefile b/drivers/watchdog/Makefile
index 0474d38..b3a6f3f 100644
--- a/drivers/watchdog/Makefile
+++ b/drivers/watchdog/Makefile
@@ -35,6 +35,9 @@ obj-$(CONFIG_USBPCWATCHDOG) += pcwd_usb.o
 
 # ALPHA Architecture
 
+# TCA hardware wdg
+obj-$(CONFIG_HW_WATCHDOG) += hw_wdg.o
+
 # ARM Architecture
 obj-$(CONFIG_ARM_SP805_WATCHDOG) += sp805_wdt.o
 obj-$(CONFIG_ARM_SBSA_WATCHDOG) += sbsa_gwdt.o
-- 
1.9.1

